dbt/models/reporting/reporting.sot_assessment_roll.py

```diff
@@ -0,0 +1,328 @@
+def clean_names(x):
+    """
+    Function to rename and reorder columns.
+    """
+
+    output = x.rename(
+        columns={
+            "av_tot_size": "pin_n_tot",
+            "av_tot_count": "pin_n_w_value",
+            "av_tot_pct_w_value": "pin_pct_w_value",
+            "geography_data_year_first": "geography_data_year",
+        }
+    )
+
+    output = output[
+        [
+            "geography_type",
+            "geography_id",
+            "geography_data_year",
+            "group_type",
+            "group_id",
+            "year",
+            "reassessment_year",
+            "stage_name",
+            "pin_n_tot",
+            "pin_n_w_value",
+            "pin_pct_w_value",
+            "av_tot_min",
+            "av_tot_q10",
+            "av_tot_q25",
+            "av_tot_median",
+            "av_tot_q75",
+            "av_tot_q90",
+            "av_tot_max",
+            "av_tot_mean",
+            "av_tot_sum",
+            "av_tot_delta_median",
+            "av_tot_delta_mean",
+            "av_tot_delta_sum",
+            "av_tot_delta_pct_median",
+            "av_tot_delta_pct_mean",
+            "av_tot_delta_pct_sum",
+            "av_bldg_min",
+            "av_bldg_q10",
+            "av_bldg_q25",
+            "av_bldg_median",
+            "av_bldg_q75",
+            "av_bldg_q90",
+            "av_bldg_max",
+            "av_bldg_mean",
+            "av_bldg_sum",
+            "av_bldg_delta_median",
+            "av_bldg_delta_mean",
+            "av_bldg_delta_sum",
+            "av_bldg_delta_pct_median",
+            "av_bldg_delta_pct_mean",
+            "av_bldg_delta_pct_sum",
+            "av_land_min",
+            "av_land_q10",
+            "av_land_q25",
+            "av_land_median",
+            "av_land_q75",
+            "av_land_q90",
+            "av_land_max",
+            "av_land_mean",
+            "av_land_sum",
+            "av_land_delta_median",
+            "av_land_delta_mean",
+            "av_land_delta_sum",
+            "av_land_delta_pct_median",
+            "av_land_delta_pct_mean",
+            "av_land_delta_pct_sum",
+        ]
+    ]
+
+    return output
+
+
+def ingest_geos(geos):
+    """
+    Function to convert dbt seed into a dictionary that can be iterated over.
+    """
+
+    geos = geos.toPandas()
+    output = {
+        k: list(geos[k].unique()[pd.notnull(geos[k].unique())])
+        for k in geos.columns
+    }
+
+    return output
+
+
+def model(dbt, spark_session):
+    """
+    Function to build a dbt python model using PySpark.
+    """
+    dbt.config(materialized="table")
+
+    # Ingest geographies and their associated data years
+    geos = ingest_geos(dbt.ref("reporting.sot_data_years"))
+
+    input = dbt.ref("reporting.sot_assessment_roll_input")
+
+    # Convert the Spark input dataframe to Pandas for
+    # compatibility with assesspy functions
+    input = input.toPandas()
+
+    df = assemble(input, geos=geos, groups=groups)
+
+    schema = (
```

Comment:
**[Thought, non-blocking]** Alternatively, if it turns out we do have to manage this schema directly, I think it would be more readable/maintainable if we defined it as a dictionary and then serialized it as a string for Spark, e.g.:

```python
schema = {
    "geography_type": "string",
    "geography_id": "string",
    "geography_data_year": "string",
    # etc.
    ...
}

spark_df = spark_session.createDataFrame(
    df,
    schema=", ".join(f"{key}: {val}" for key, val in schema.items())
)
```